### ISSUE TITLE
csv and shapefile uploads

### ISSUE THREAD

**Issue by Vince-Egan** [kartoza/LEDET_BIMS#265]
Could we please have a field for 'owner' (data owner) for both csv and shapefile bulk data uploads - we'll need this to sort LEDET data from SANBI and other third-party data.

**Comment by Vince-Egan** [comment]
I've just attempted to format a shapefile for data upload - 3 of the mandatory attribute field-names are longer than the 8 characters allowed for shapefiles, making it impossible to upload in this format.

**Comment by dimasciput** [comment]
@Vince-Egan the required field for shapefiles is different from csv,
`species(string), location(string), notes(string), category(string), date(yyyy-mm-dd), collector(string)`. 

For the owner, currently the uploader will be the owner of the data. Would you like the owner as a string field?

**Comment by Vince-Egan** [comment]
@dimasciput We need owner as a string so that we can separate third-party data from LEDET data. We will be uploading data from SANBI that must be filtered out when SANBI retrieves data from the system. The default value here would be 'LEDET' rather than a specific official. An alternative would be to have a boolean field that marks the record as LEDET-owned?

**Comment by gubuntu** [comment]
this is two separate issues

1. @dimasciput, shapefile dbfs have a field name limit of 10 characters so if they are shorter than the model fields then the app needs to map them appropriately. csv field names on the other hand can be exactly the same as the model field names. 
1. There are at least three people or orgs associated with a record: the person who uploaded it (let's call that the 'owner' or contributor), the observer / collector and the custodian (LEDET, SANBI or a person). We should use Darwin Core fields to capture all of these if there is a mapping. The richer the biodiversity metadata the better, since we don't have the rich CSW metadata for biodiversity data like we do for the GIS layers in GeoNode.

**Comment by dimasciput** [comment]
@gubuntu 
> shapefile dbfs have a field name limit of 10 characters so if they are shorter than the model fields then the app needs to map them appropriately. csv field names on the other hand can be exactly the same as the model field names.

This is already handled in testing server, the required headers for shapefile are shorter

> There are at least three people or orgs associated with a record: the person who uploaded it (let's call that the 'owner' or contributor), the observer / collector and the custodian (LEDET, SANBI or a person). We should use Darwin Core fields to capture all of these if there is a mapping. The richer the biodiversity metadata the better, since we don't have the rich CSW metadata for biodiversity data like we do for the GIS layers in GeoNode.

Currently we have owner and collector field in our model, so we add custodian field for @Vince-Egan needs ?

**Comment by gubuntu** [comment]
yes - is there not such a field anyway in Darwin core?

**Comment by dimasciput** [comment]
I'm not sure with those fields, I just heard of it now. Is it similar with these fields from taxon models -> https://github.com/kartoza/django-bims/blob/develop/bims/models/taxon.py#L41

**Comment by gubuntu** [comment]
institutionID seems to be the closest field for custodian in DarwinCore http://rs.tdwg.org/dwc/terms/index.htm#georeferenceSources

For other field mappings poke around that site

**Comment by gubuntu** [comment]
why was this closed? pls comment when closing. was institutionID implemented?

**Comment by dimasciput** [comment]
@gubuntu yes, implemented, and for ledet site the default institutionId is ledet.
